duet/api.py

```diff
@@ -435,8 +443,9 @@ async def acquire(self) -> "Slot":
     async def __aexit__(self, exc_type, exc, tb) -> None:
         self._release()
 
-    def _release(self):
-        self._count -= 1
+    def _release(self, decrement: bool = True) -> None:
+        if decrement:
+            self._count -= 1
```

Comment:
I originally wanted to do that, but because the Limiter itself is a context manager (it has `__aenter__` and `__aexit__` methods), we don't have a way to communicate something like a token from one _particular_ `__aenter__` call to the matching `__aexit__` call, so we can't play the same trick as we do for the scheduler where we can store a token in the locals of an async generator function which serves as a context manager. We did that here because we can avoid allocating a context manager object in the uncontended case, but it means here that we have to be "stateless" and can't communicate information to the `_release` call to know which future corresponds to this particular call.

Reply:
Oh I see; in the future, maybe we could have `__aenter__` return a _new_ object that wraps the token? But yeah I see your point.